jetty-core/jetty-http2/jetty-http2-client-transport/src/main/java/org/eclipse/jetty/http2/client/transport/internal/HttpReceiverOverHTTP2.java

```diff
@@ -261,7 +261,31 @@ public Runnable onTimeout(TimeoutException failure, Promise<Boolean> promise)
     @Override
     public Runnable onFailure(Throwable failure, Callback callback)
     {
-        Promise<Boolean> promise = Promise.from(failed -> callback.succeeded(), callback::failed);
-        return new Invocable.ReadyTask(Invocable.InvocationType.NON_BLOCKING, () -> responseFailure(failure, promise));
+        HttpExchange exchange = getHttpExchange();
+        if (exchange != null)
+            return new FailureTask(exchange, failure, callback);
+        callback.succeeded();
+        return null;
+    }
+
+    private static class FailureTask extends Invocable.Task.Abstract
```

Comment:
Did you create this class to have nice-looking names in the debug logs? In this case it may be worth stating that in a comment.